enhancements/ignition-spec-dual-support.md

```diff
@@ -0,0 +1,327 @@
+existing spec 2 configs to spec 3. The translation will happen as soon as the
+support for ignition spec 3 lands in MCO. This means for existing machines,
+the existing rendered machineconfig will be translated to ignition spec 3 in
+the MCC. The MCS will have additional functionality to translate the existing
+ignition configs to spec 3. The MCS will also be able to detect which ignition
+version to serve a newly booted image for initial provisioning of the system.
+Then it will update the new node to align with the rest of the cluster, using
+spec 3. New machineconfigs applied to the cluster that has an ignition snippet
+on spec 2 should no longer be allowed. 
+
+The MCO will also throw an alert upon:
+ - An attempted to update to the new ignition spec cannot be performed due to untranslatable configs, and the user must manually update/remove the untranslatable config before the update can continue
+ - The cluster sees a spec 2 machineconfig applied to it after already updating to spec 3, and rejects that machineconfig.
+
+The MCO should also add ability to reconcile broken spec 2 -> spec 3 updates,
+after manual intervention from the user.
+
+
+### User Stories
+
+** As the admin of an existing 4.x cluster on spec 2, I’d like to upgrade to the newest version and use ignition spec 3 **
+
+Acceptance criteria:
+ - The update completes without user intervention, if all machine configs existing on the cluster can be directly translated to spec 3
+ - The user receives an alert, if the update is unable to complete due to untranslatable configs
+ - The user is able to recover the cluster and finish the update if the untranslatable configs are manually translated or remove, or roll back to the old version
+ - The user should have received notification that the update will be changing spec version, as well as received necessary documentation on how to recover a failed update
+ - CI tests are put in place to make sure the existing versions can be updated to the new payload
+
+** As a user of openshift, I’d like to install from a spec 2 bootimage andimmediately update to a spec 3 payload **
+
+Acceptance criteria:
+ - Essentially the same as story 1
+
+** As a user of Openshift, I’d like to install a fresh ignition spec 3 cluster **
+
+Acceptance criteria:
+ - The workflow remains the same for an IPI cluster
+ - The workflow remains the same for a UPI cluster, minus custom specification changes
+ - The user should have good documentation, based on version, of how to set up user defined configs during install time
+
+** As an admin of an existing spec 3 cluster, I’d like to apply a new machineconfig **
+
+Acceptance criteria:
+ - The machineconfig is applied successfully, if the user has defined a correct spec 3 ignition snippet
+ - The user is properly alerted if they attempt to apply a spec 2 config, and the machineconfig fails to apply
+ - The user is given necessary docs to remove the undesired spec 2 config and to translate it to a spec 3 config
```

Comment:
What is "the undesired spec 2 config" in this case? The last bullet says any spec 2 config would fail to apply.

Reply:
Sorry, it just means any spec 2 config in the system. Basically steps to `oc delete machineconfig/xxx` and then apply a new one